### ISSUE TITLE
ocamldep: Argument list too long

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7011]
**Original bug ID:** 7011
**Reporter:** dhekir
**Assigned to:** @mshinwell
**Status:** resolved (set by @mshinwell on 2016-12-12T16:29:20Z)
**Resolution:** fixed
**Priority:** normal
**Severity:** feature
**Version:** 4.02.1
**Target version:** 4.05.0 +dev/beta1/beta2/beta3/rc1
**Category:** tools (ocaml{lex,yacc,dep,debug,...})
**Child of:** #7050
**Monitored by:** @gasche @diml "Julien Signoles"

## Bug description

ocamldep expects all arguments to be given in the command line, but when there are too many of them, and the ARG_MAX value is low (e.g. 32-bit Cygwin has it set to 32000), it results in an "Argument list too long" error.

I couldn't find a workaround for this issue. It does not seem that I could split the argument list in several commands without altering the result. If there's a way to do so, I'd be interested to learn it.

Otherwise, adding an option such as "-f <input file>" could avoid this issue by allowing the user to add filenames one by one into a file, and then passing the file to ocamldep.

**Comment by vicuna** [comment]
**Comment author:** @xavierleroy

See #7050 for work in progress on something similar to your "-f <file>".

This said, what is wrong with 

   xargs ocamldep [-I options] < listoffiles

?

**Comment by vicuna** [comment]
**Comment author:** dhekir

Indeed, it is likely that xargs would work fine. I cannot reliably reproduce my environment now to test it again, but it should work.

My issue was that, because of the stack limitation under Cygwin, I tried emulating the same behavior using xargs on my Linux using ulimit -s 32, and xargs reported the same issue there ("Argument list too long"), so I assumed it would not work.

I just found out about xargs' "-s <buffer size>" argument (which I should have used on Cygwin as well), and it should allow xargs to properly construct the input without any problems.

Anyway, as soon as I am able to reconfigure my environment to retry it, I will report back. Otherwise, the -args feature of 0007050 should also work for me.
